### ISSUE TITLE
Writer's guide - Types notation

### ISSUE THREAD

**Issue by montogeek** [webpack/webpack.js.org#3333]
Some time ago a writes guide was created that specifies how to add types notations to our documentation, but later some users found it confusing, an alternative is to use TypeScript notation which has better standardization process and it is more known by the community.

Another alternative is to use TypeScript documentation from webpack source code directly (it would be a fun AST ride, maybe).

**Comment by montogeek** [comment]
Related: https://github.com/webpack/webpack.js.org/pull/3323

**Comment by EugeneHlushko** [comment]
If this doesnt get traction, i suggest adding a small outline to the types in the docs somewhere or as i suggested earlier, a tooltip when hover types

**Comment by rishabh3112** [comment]
@EugeneHlushko Typescript notations are less confusing and should be used in my opinion.

> Another alternative is to use TypeScript documentation from webpack source code directly (it would be a fun AST ride, maybe).

Major part of the code in the webpack source code in written in javascript with commented types, thus extracting the types from there wouldn't be a nice idea.

**Comment by chenxsan** [comment]
I agree with @rishabh3112 here. We can see the two version here https://github.com/webpack/webpack.js.org/pull/3706#issuecomment-619119074, the TypeScript one is much better. Also, when user reading docs, they won't read the writers guide first to figure out the difference. So it would be better to choose the popular notations.

We might use [`details`/`summary`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/details) here:

<details>
  <summary>Click to see Options types</summary>
<pre>
type Entry =
  | string
  | [string]
  | {
      key:
        | string
        | [string]
        | {
            import: string | [string]
            dependOn: string | [string]
            filename: string
          }
    }
  | (() =>
      | string
      | [string]
      | {
          key:
            | string
            | [string]
            | {
                import: string | [string]
                dependOn: string | [string]
                filename: string
              }
        })
</pre>
</details>
